x-pack/plugins/security_solution/server/lib/detection_engine/prebuilt_rules/logic/diff/calculation/algorithms/scalar_array_diff_algorithm.ts

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { difference, union } from 'lodash';
+import { assertUnreachable } from '../../../../../../../../common/utility_types';
+import type {
+  ThreeVersionsOf,
+  ThreeWayDiff,
+} from '../../../../../../../../common/api/detection_engine/prebuilt_rules';
+import {
+  determineOrderAgnosticDiffOutcome,
+  determineIfValueCanUpdate,
+  ThreeWayDiffOutcome,
+  ThreeWayMergeOutcome,
+  MissingVersion,
+} from '../../../../../../../../common/api/detection_engine/prebuilt_rules';
+
+/**
+ * Diff algorithm used for arrays of scalar values (eg. numbers, strings, booleans, etc.)
+ *
+ * NOTE: Diffing logic will be agnostic to array order
+ */
+export const scalarArrayDiffAlgorithm = <TValue>(
```

Comment:
Nit: For now, we could restrict elements to be either strings or numbers. There's a chance we'll never need to support other primitive types.

```suggestion
export const scalarArrayDiffAlgorithm = <TValue = string | number>(
```